### PR TITLE
Fixes all statement leaks in the driver.

### PR DESCRIPTION
**Pull request by peterbae**
Close all statements in the driver that are initialized locally but are never / sometimes not closed.

Fixes issue #308.

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java [hunk 1]
```diff
@@ -244,12 +244,12 @@ private SQLServerResultSet getResultSetFromInternalQueries(String catalog,
         SQLServerResultSet rs = null;
         try {
             rs = ((SQLServerStatement) connection.createStatement()).executeQueryInternal(query);
-
         }
         finally {
-            if (null != orgCat) {
+            if (null != rs)
+                rs.close();
```
Comment (v-xiangs) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: this method returns the resultset, so I guess we should not close this resultset here

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 2]
```diff
@@ -1035,6 +1036,10 @@ else if (resultSet != null) {
                 SQLServerException.makeFromDriverError(connection, this, form.format(msgArgs), null, true);
             }
         }
+        finally {
+            if (null != stmt)
+                stmt.close();
```
Comment (v-xiangs) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: this statement creates resultset which will be returned, closing this statement also close the returned resultset, which is not what we want I guess

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java [hunk 3]
```diff
@@ -244,12 +244,10 @@ private SQLServerResultSet getResultSetFromInternalQueries(String catalog,
         SQLServerResultSet rs = null;
         try {
             rs = ((SQLServerStatement) connection.createStatement()).executeQueryInternal(query);
-
         }
         finally {
-            if (null != orgCat) {
+            if (null != orgCat)
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: I don't think we ever put brackets when we could omit them. so i omit them here as well

Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: I think we always omit the bracket when we can, so i did the same here.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParameterMetaData.java [hunk 4]
```diff
@@ -563,6 +563,8 @@ private void checkClosed() throws SQLServerException {
         assert null != st;
         stmtParent = st;
         con = st.connection;
+        SQLServerStatement s = null;
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParameterMetaData.java]: i guess s doesnt need to be instantiated to null but i did it for consistency

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 5]
```diff
@@ -98,6 +98,9 @@
 
     /** The prepared statement handle returned by the server */
     private int prepStmtHandle = 0;
+    
+    /** Statement used for getMetadata(). Declared as a field to facilitate closing the statement. */
+    private SQLServerStatement internalStmt = null;
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: having internalStmt as a field is the only way to close it after the user is done working with the ResultsetMetadata.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java (hunk 3 above)
Comment (v-afrafi) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: actually we started adding back the brackets even with one line of code, so lets keep the brackets here  😄

Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: done.

**Pull request merged**